src/vcpkg/base/curl.cpp

```diff
@@ -0,0 +1,124 @@
+#include <vcpkg/base/checks.h>
+#include <vcpkg/base/curl.h>
+
+namespace
+{
+    struct CurlGlobalInit
+    {
+        CurlGlobalInit() : init_status(curl_global_init(CURL_GLOBAL_DEFAULT)) { }
+        ~CurlGlobalInit() { curl_global_cleanup(); }
+
+        CurlGlobalInit(const CurlGlobalInit&) = delete;
+        CurlGlobalInit(CurlGlobalInit&&) = delete;
+        CurlGlobalInit& operator=(const CurlGlobalInit&) = delete;
+        CurlGlobalInit& operator=(CurlGlobalInit&&) = delete;
+
+        CURLcode get_init_status() const { return init_status; }
+
+    private:
+        CURLcode init_status;
+    };
+}
+
+namespace vcpkg
+{
+    CURLcode get_curl_global_init_status() noexcept
+    {
+        static CurlGlobalInit g_curl_global_init;
+        return g_curl_global_init.get_init_status();
+    }
+
+    CurlEasyHandle::CurlEasyHandle() : m_ptr(nullptr) { vcpkg::get_curl_global_init_status(); }
```

Comment:
```suggestion
    CurlEasyHandle::CurlEasyHandle() { get_curl_global_init_status(); }
```

pointer already initialized in header.